src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2412,9 +2465,30 @@ void visitArgumentEvaluateAndUnsplit(int argumentIndex, bool assertsTrue, bool a
                         bool reportedWarning = false;
                         if (argument.Kind != BoundKind.SuppressNullableWarningExpression)
                         {
-                            reportedWarning = ReportNullableAssignmentIfNecessary(argument, parameterType, resultType, useLegacyWarnings: false, assignmentKind: AssignmentKind.Argument, target: parameter) ||
-                                ReportNullableAssignmentIfNecessary(argument, resultType, parameterType, useLegacyWarnings: UseLegacyWarnings(argument));
+                            TypeSymbolWithAnnotations adjustedResultType = resultType;
+
+                            if (adjustedResultType.IsPossiblyNullableReferenceTypeTypeParameter() && !parameterType.IsPossiblyNullableReferenceTypeTypeParameter())
+                            {
+                                // This code path is likely not reachable due to https://github.com/dotnet/roslyn/issues/30946.
+                                // See unit test NullabilityOfTypeParameters_080 for an attempt to get here.
+                                adjustedResultType = TypeSymbolWithAnnotations.Create(adjustedResultType.TypeSymbol, NullableAnnotation.NullableBasedOnAnalysis);
+                            }
```

Comment:
This code snippet is used three times in this method. Consider extracting a local function.

Reply:
> This code snippet is used three times in this method. Consider extracting a local function.

Will do

---
In reply to: [232116712](https://github.com/dotnet/roslyn/pull/30913#discussion_r232116712) [](ancestors = 232116712)